src/NUnitFramework/tests/Assertions/AssertWarnTests.cs

```diff
@@ -51,5 +52,47 @@ public void AssertWarnWorksWithMessageAndArgs()
             Assert.AreEqual(ResultState.Warning, result.ResultState);
             Assert.AreEqual("MESSAGE: 2+2=4", result.Message);
         }
+
+        [Test]
+        public void WarningsAreDisplayedWithFailure()
+        {
+            ITestResult result = TestBuilder.RunTestCase(
+                typeof(WarningFixture),
+                "TwoWarningsAndFailure");
+
+            Assert.That(result.ResultState, Is.EqualTo(ResultState.Failure));
+            Assert.That(result.AssertionResults.Count, Is.EqualTo(3));
```

Comment:
Similarly here, don't we have an assertion to check collection counts directly, `Has.Exactly`?

Reply:
My personal style is to try to limit myself to the most basic assertions when testing NUnit. Otherwise, I have to worry about what I'm testing in order to avoid using the feature that I am currently testing.

Reply:
> My personal style is to try to limit myself to the most basic assertions when testing NUnit. Otherwise, I have to worry about what I'm testing in order to avoid using the feature that I am currently testing.

Now this is interesting and I see where you are coming from. Going by that logic, I'd even consider using exclusively the barebones `Assert.IsTrue`.

Reply:
So it's a balance between good messages and simplicity. To have good messages with IsTrue you have to add a message with each assert.

As a matter of general interest, the nunit 3 syntax came from nunitlite, which I boots trapped with about a half dozen assertions.